### PR TITLE
Experimental pub/sub implementation

### PR DESCRIPTION
**Pull request by bwlewis**
This patch defines a minimalist pub/sub implementation.

### CODE REVIEW

File: R/pubsub.R [hunk 1]
```diff
@@ -1,46 +1,23 @@
...
+# Callback handler for convenience
+redisMonitorChannels <- function(context)
 {
-  x <- .getResponse()
-  if(length(x)!=3 && x[[1]] != "message") return(x)
+  x <- listen(context)
+  if(length(x)!=3 || x[[1]] != "message") return(x)
   if(exists(x[[2]],mode="function")) {
     return(do.call(x[[2]],as.list(x[[3]])))
   }
```
Comment (eddelbuettel) [inline R/pubsub.R:23]: I am wondering if we should/could also allow for a non-function return.  Maybe as simple as a last statement `x[[3]]` as a fallback?

**Comment by eddelbuettel** [comment]
This is really awesome and greatly simplifies the use.

I was wondering if I could ask you to just slab a copyright statement in for the files this adds / modifies?  Because this now no longer imports such changes from `rredis` but rather extends what was in `RcppRedis` -- and does it so elegantly and in a way I did not see when I looked at this last year -- that we can maybe just stick with the GPL (>= 2) this hand?

**Comment by bwlewis** [comment]
Yeah, GPL is just fine!...won't the new files (only man pages and pubusb.R) automatically be covered by the package license? I think that's probably enough...

On another note, it's a draft. It works but can be improved, I worry about use of exec and hiredis...

**Comment by eddelbuettel** [comment]
Yes it probably is fine -- and we have your nod of record here now too which should be good enough.

And "draft" and permanent beta (in the Google sense) is perfectly fine by me.  Perfection is for the afterlife.

File: R/pubsub.R (hunk 1 above)
Comment (eddelbuettel) [reply R/pubsub.R:23]: Or maybe not.  I just updated my example by moving two more statements into the callback used here, and it is bloody elegant.  Really like this.

**Review by eddelbuettel — APPROVED** [review]
Lovely stuff, mergin

**Comment by eddelbuettel** [comment]
FYI I did three minor post-merge edits now pushed too
- there was a dangling unused `SEXP` var in `getreply` which I removed
- the `getreply` and `push` code was moved to the end of the file because sort is (somewhat incomprehensively) most chronological (as topics got added)
- could not resist renaming `getreply` to `getReply` for consistency in `Redis.cpp` and `pubsub.R`

**Pull request merged**